packages/medusa/src/services/fulfillment.ts

```diff
@@ -224,11 +234,34 @@ class FulfillmentService extends TransactionBaseService {
 
       const created = await Promise.all(
         fulfillments.map(async ({ shipping_method, items }) => {
+          await pvInventoryServiceTx.validateInventoryAtLocation(
+            items,
+            locationId!
+          )
+
+          await Promise.all(
+            items.map(async (i) => {
+              await pvInventoryServiceTx.adjustReservationsQuantityByLineItem(
+                i.id,
+                i.variant_id!,
+                locationId!,
+                -i.quantity
+              )
+
+              await pvInventoryServiceTx.adjustInventory(
+                i.variant_id!,
+                locationId!,
+                -i.quantity
+              )
```

Comment:
thought (non-blocking): we may have to create reservations as part of a data migration script when users turn on the inventory module.

The case I can think of is:

Order A is placed - this reduces the inventory_quantity
User turns on the inventory module
Order A gets fulfilled - this reduces the inventory level's quantity
Outcome: the stock for Order A has been accounted for twice.

If we create reservations for all orders that are placed but not fulfilled that should circumvent this.